tuist/tuist
# Fix linting rules for allowing a watchOS as a dependency of a test target

Resolves https://github.com/tuist/tuist/issues/4907

### Short description 📝

This PR should fix #4907 by enabling the combination of `product: .app, platform: .watchOS` to be valid when included in an `.unitTest` and `.uiTest` target. This should fix issues when the new way of using a watchOS app target, introduced in this PR https://github.com/tuist/tuist/pull/4658 is used. 

I am not sure how we should handle older Xcode versions here, but I guess it should not be possible to generate a project with only a `product: .app, platform: .watchOS` target in, with Xcode versions lower than 14.0. But I would love to hear your opinion about that. 

### How to test the changes locally 🧐

1. Checkout  https://github.com/alexanderwe/tuist-watchos-unit-test-host-app-issue
2. Run `tuist generate`
3. Verify generation works 
4. Verify watchOS host app is added correctly to `FrameworkTests_watchos` target 
5. Verify tests are running 

### Checklist ✅
